### ISSUE TITLE
declare of sk_alloc was changed from kernel 4.2.

### ISSUE THREAD

**Issue by nizq** [pfq/PFQ#19]
```
runhaskell Build.hs install  --sandbox=shared-sandbox
[1/14] install pfq.ko:
.[1/1] build pfq.ko:
..[1/1] configure pfq.ko:
make -C /lib/modules/4.3.0-1-amd64/build M=/home/nizq/workspace/github/pfq/kernel modules
make[1]: Entering directory '/usr/src/linux-headers-4.3.0-1-amd64'
Makefile:10: *** mixed implicit and normal rules: deprecated syntax
  CC [M]  /home/nizq/workspace/github/pfq/kernel/pf_q.o
/home/nizq/workspace/github/pfq/kernel/pf_q.c: In function ‘pfq_create’:
/home/nizq/workspace/github/pfq/kernel/pf_q.c:798:14: error: too few arguments to function ‘sk_alloc’
         sk = sk_alloc(net, PF_INET, GFP_KERNEL, &pfq_proto);
              ^
In file included from /home/nizq/workspace/github/pfq/kernel/pf_q.c:47:0:
/usr/src/linux-headers-4.3.0-1-common/include/net/sock.h:1492:14: note: declared here
 struct sock *sk_alloc(struct net *net, int family, gfp_t priority,
              ^
/usr/src/linux-headers-4.3.0-1-common/scripts/Makefile.build:263: recipe for target '/home/nizq/workspace/github/pfq/kernel/pf_q.o' failed
make[4]: *** [/home/nizq/workspace/github/pfq/kernel/pf_q.o] Error 1
/usr/src/linux-headers-4.3.0-1-common/Makefile:1395: recipe for target '_module_/home/nizq/workspace/github/pfq/kernel' failed
make[3]: *** [_module_/home/nizq/workspace/github/pfq/kernel] Error 2
Makefile:146: recipe for target 'sub-make' failed
make[2]: *** [sub-make] Error 2
Makefile:8: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-4.3.0-1-amd64'
Makefile:38: recipe for target 'all' failed
make: *** [all] Error 2
SimpleBuilder: build pfq.ko aborted: 'make' command failed!
```

----------------------------------------------------------------------------------------------------------------------

version <= 4.1:
```
struct sock *sk_alloc(struct net *net, int family, gfp_t priority,
                                  struct proto *prot);
```

version > 4.1:
```
struct sock *sk_alloc(struct net *net, int family, gfp_t priority,
                                  struct proto *prot, int kern);
```

**Comment by awgn** [comment]
Thanks for reporting. I'll fix this for kernel 4.2 in the upcoming release!

Regards,
N.

**Comment by awgn** [comment]
pfq.ko should now compile with kernel 4.2.
